coralproject/talk
# Design Pass Admin

# Design Pass Admin

Consider: 
- `react-mdl` is hard to override
- Avoid using `react-mdl`. Use `coral-ui` instead. 
- If that component doesn’t exist, let’s create it! :)

## New UI components
- List
  - Item
- Card
- Pager

Old Code without using `coral-ui` and overriding `react-mdl`
```js
            <List>
              <ListItem className={`${styles.settingOption} ${activeSection === 'comments' ? styles.active : ''}`}>
                <ListItemContent
                  onClick={this.changeSection('comments')}
                  icon='settings'>{lang.t('configure.comment-settings')}</ListItemContent>
              </ListItem>
              <ListItem className={`${styles.settingOption} ${activeSection === 'embed' ? styles.active : ''}`}>
                <ListItemContent
                  onClick={this.changeSection('embed')}
                  icon='code'>{lang.t('configure.embed-comment-stream')}</ListItemContent>
              </ListItem>
              <ListItem className={`${styles.settingOption} ${activeSection === 'wordlist' ? styles.active : ''}`}>
                <ListItemContent
                  onClick={this.changeSection('wordlist')}
                  icon='settings'>{lang.t('configure.wordlist')}</ListItemContent>
              </ListItem>
            </List>

```


Beautiful usage of `coral-ui`

```js
            <List onChange={this.changeSection}>
              <Item itemId='comments' icon="settings">
                {lang.t('configure.comment-settings')}
              </Item>
              <Item itemId='embed' icon='code'>
                {lang.t('configure.embed-comment-stream')}
              </Item>
              <Item itemId='wordlist' icon='settings'>
                  {lang.t('configure.wordlist')}
              </Item>
            </List>
```

## How to test this?

- Fetch: `git fetch`
- Switch to `design-pass-adm` branch `git checkout design-pass-adm`
- Log in as Admin
- Go to `/admin`
- Check the functionality 

### Note:
After merging this, @IAmSamHankins and @egrdina can take a look and give feedback
